turbopack/crates/turbopack-core/src/resolve/mod.rs

```diff
@@ -2450,7 +2485,18 @@ async fn resolve_import_map_result(
             {
                 None
             } else {
-                Some(resolve_internal(lookup_path, request, options))
+                let result = resolve_internal(lookup_path, request, options);
+                if let (Some(rewritten), Some(original)) = (
+                    request.await?.request_pattern(),
+                    original_request.await?.request_pattern(),
+                ) {
+                    Some(result.with_replaced_request_key_pattern(
+                        Value::new(rewritten),
+                        Value::new(original),
+                    ))
+                } else {
+                    Some(result)
+                }
```

Comment:
This is where the `RequestKey` rewriting happens now when using import maps, so that the runtime lookup works correctly.
- The returned request key might be `./messages/en.json` 
- The `rewritten` pattern would be `./messages/DYNAMIC.json`
- The `original` pattern would be `@/messages/DYNAMIC.json`
- Then the resulting key is `@/messages/en.json`